### PR TITLE
fix: Made Modal accessible

### PR DESCRIPTION
**Pull request by GoOz**
Refs: #601

### CODE REVIEW

File: react/Modal/index.jsx [hunk 1]
```diff
@@ -158,7 +161,12 @@ const ModalHeader = ({
 }) => {
   const isTitle = typeof children === 'string'
   return (
-    <div className={cx(styles['c-modal-header'], className)} style={style}>
+    <div
+      className={cx(styles['c-modal-header'], className)}
+      style={style}
+      id={titleId}
+      tabIndex="-1"
```
Comment (GoOz) [inline react/Modal/index.jsx]: `tabIndex="-1"` is to make sure that when the modal takes the focus, reading began at the beginning of the modal and not the next focusable element (inside the modal) which is the behaviour by default.

Comment (ptbrowne) [reply react/Modal/index.jsx]: Why do we have to put an id ?

Comment (GoOz) [reply react/Modal/index.jsx]: In order to assign a title to the modal, having a `<hn>` isn't always enough apparently.

**Comment by drazik** [comment]
Is there plans to trap the focus inside the `Modal` when the user opens it ? Currently we can tab behind it.

**Comment by GoOz** [comment]
@drazik according to what I read, the fact that the modal has `role="dialog"` and `aria-modal=true` is enough to give the focus and ignoring the background content. Before `aria-modal=true` from WAI 1.1, you had to add `aria-hidden` on content behind to ensure this behaviour, but not anymore.

File: react/Modal/index.jsx (hunk 1 above)
Comment (ptbrowne) [reply react/Modal/index.jsx]: Assigning a title, what for ?

Comment (GoOz) [reply react/Modal/index.jsx]: It's a label, to be read, so the reader knows what the content of the modal is all about.

Comment (GoOz) [reply react/Modal/index.jsx]: There's `aria-labelledBy={titleId}` on the Modal, L320
I didn't add `aria-describedby` as it is optional, contextual and in many case, not relevant.
As for the unique ID being generated outside the modal, if I generate it inside the Modal, how can `ModalHeader` can use it too then?

Comment (ptbrowne) [reply react/Modal/index.jsx]: I did not understand that aria-labelledBy was a reference and not the text to be read.

`titleId` should be generated in the Modal render and passed to ModalHeader though.

**Comment by drazik** [comment]
Oh, OK :) Thanks @GoOz

File: react/Modal/Readme.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ initialState = { modalDisplayed: false};
   <button onClick={()=>setState({ modalDisplayed: !state.modalDisplayed })}>
     Toggle modal
   </button>
-  {state.modalDisplayed && <Modal  description={content.ada.short} dismissAction={() => setState({ modalDisplayed: false })} /> }
+  {state.modalDisplayed && <Modal description={content.ada.short} arial-title="Accessible title" dismissAction={() => setState({ modalDisplayed: false })} /> }
```
Comment (ptbrowne) [inline react/Modal/Readme.md]: arial -> aria

Comment (GoOz) [reply react/Modal/Readme.md]: Wow that's a 100% pure *typo* 😆
Nice catch, thanks.

**Pull request merged**